File: src/main/java/org/zenbot/szolnok/timetable/batch/batch/step/bus/processor/StringCleaner.java
package org.zenbot.szolnok.timetable.batch.batch.step.bus.processor;

import org.springframework.stereotype.Component;

@Component
public class StringCleaner {
    public String clean(String string) {
        return string
                .replaceAll("û", "ű")
                .replaceAll("õ", "ő")
                .replaceAll("ô", "ő");
    }
}

File: src/main/kotlin/org/zenbot/szolnok/timetable/batch/step/bus/processor/StringCleaner.kt
package org.zenbot.szolnok.timetable.batch.step.bus.processor

import org.springframework.stereotype.Component

@Component
class StringCleaner {
    fun clean(string: String): String {
        return string
                .replace("û".toRegex(), "ű")
                .replace("õ".toRegex(), "ő")
                .replace("ô".toRegex(), "ő")
                .replace("\\.".toRegex(), "")
                .trim { it <= ' ' }
                .replace(" +".toRegex(), " ")
    }
}
